front: fix osm track and 3d layers never displayed

Synar opened the pull request:
Close #13421
Close #13423

The osm track and osm layer components were not passed the boolean props toggling on the osm track and 3d building layers.

Fix this, and make the props non optional to avoid this issue popping up again.

Synar committed:
front: fix osm track and 3d layers display never activated

The osm track and osm layer components were not passed the boolean props
toggling on the osm track and 3d building layers.

Fix this, and make the props non optional to avoid this issue popping up again.

Signed-off-by: Alice Khoudli <alice.khoudli@polytechnique.org>

sim51 commented:
For the OSM track layer, it works!

But for the 3D buildings, I don't see it, but perhaps I missed something ?

<img width="1891" height="987" alt="image" src="https://github.com/user-attachments/assets/d33aa61a-fb6c-4207-b280-9a6765393629" />

<img width="1891" height="987" alt="image" src="https://github.com/user-attachments/assets/4f53aca1-ae58-41a7-bcc7-7c22b48f3d50" />

sim51 commented:
Can you add this layer in the file `front/src/common/Map/theme/OSMStyle.ts` just after the `building` one (at line 291) :

```
{
      id: 'building-3d',
      type: 'fill-extrusion',
      source: 'openmaptiles',
      'source-layer': 'building',
      minzoom: 17,
      layout: {
        visibility: 'visible',
      },
      paint: {
        'fill-extrusion-color': 'rgba(239, 236, 236, 1)',
        'fill-extrusion-height': {
          type: 'identity',
          property: 'render_height',
        },
        'fill-extrusion-base': {
          property: 'render_min_height',
          type: 'identity',
        },
        'fill-extrusion-opacity': 0.8,
      },
    },
   ```

That solves the 3D building issue (my bad, I removed it when I did the map style refacto).

Synar committed:
front: fix 3d building style missing

Signed-off-by: Alice Khoudli <alice.khoudli@polytechnique.org>
Co-authored-by: Benoit Simard <contact@bsimard.com>

Synar commented:
> Can you add this layer in the file `front/src/common/Map/theme/OSMStyle.ts` just after the `building` one (at line 291) :
> 
> That solves the 3D building issue (my bad, I removed it when I did the map style refacto).

Thanks a lot! I did not catch this, since I was testing directly on top of romain's pr, which was before your changes.

The 3d layer only used to be present for the "minimal" style, which I assume is why it got accidentally dropped from your pr. With your proposed change, it would be present for all styles - though this does seem better, since the user has to actively toggle it on anyway. 

I added the changes to a new commit!

sim51 commented:
Tested & approved on my side.
Thanks!

SharglutDev reviewed (approved):
Lgtm and tested ✅
